src/checkout-buttons/strategies/paypal-button-strategy.ts

```diff
@@ -0,0 +1,142 @@
+            return super.initialize(options);
+        }
+
+        const paypalOptions = options.paypal;
+        const state = this._store.getState();
+        const paymentMethod = this._paymentMethod = state.paymentMethods.getPaymentMethod(options.methodId);
+
+        if (!paypalOptions) {
+            throw new InvalidArgumentError();
+        }
+
+        if (!paymentMethod || !paymentMethod.config.merchantId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        const env = paymentMethod.config.testMode ? 'sandbox' : 'production';
+        const clientToken: PaypalClientToken = { [env]: paypalOptions.clientId };
+
+        return this._paypalScriptLoader.loadPaypal()
+            .then(paypal => {
+                const allowedSources = [];
+                const disallowedSources = [];
+
+                if (paypalOptions.allowCredit) {
+                    allowedSources.push(paypal.FUNDING.CREDIT);
+                } else {
+                    disallowedSources.push(paypal.FUNDING.CREDIT);
+                }
+
+                return paypal.Button.render({
+                    env,
+                    client: clientToken,
+                    commit: paypalOptions.shouldProcessPayment,
+                    funding: {
+                        allowed: allowedSources,
+                        disallowed: disallowedSources,
+                    },
+                    style: {
+                        shape: 'rect',
+                        ...pick(paypalOptions.style, 'layout', 'size', 'color', 'label', 'shape', 'tagline', 'fundingicons'),
+                    },
+                    payment: (data, actions) => this._setupPayment(paymentMethod.config.merchantId, actions, paypalOptions.onPaymentError),
+                    onAuthorize: (data, actions) => this._tokenizePayment(data, actions, paypalOptions.shouldProcessPayment, paypalOptions.onAuthorizeError),
+                }, options.containerId);
+            })
+            .then(() => super.initialize(options));
+    }
+
+    deinitialize(): Promise<void> {
+        if (!Object.keys(this._isInitialized).length) {
+            return super.deinitialize();
+        }
+
+        this._paymentMethod = undefined;
+
+        return super.deinitialize();
+    }
+
+    private _setupPayment(merchantId: string | undefined, actions?: PaypalActions, onError?: (error: StandardError) => void): Promise<string> {
+        return this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout())
+            .then(state => {
+                const checkout = state.checkout.getCheckout();
+                const config = state.config.getStoreConfig();
+
+                if (!actions) {
+                    throw new NotInitializedError(NotInitializedErrorType.CheckoutButtonNotInitialized);
+                }
+
+                if (!checkout) {
+                    throw new MissingDataError(MissingDataErrorType.MissingCheckout);
```

Comment:
Can you see my comment above? If you're not using the object, you probably don't need to do a null check here as well.